test/run-drun/idx.mo

```diff
@@ -0,0 +1,27 @@
+actor a {
+    type O1 = { field : Int }; // "\ba\94\93\00"
+    type O2 = { a: Int; field : Int }; // "a\00\00\00\ba\94\93\00"
+    type O3 = { a: Int; field : Int; other : Int }; // "a\00\00\00\ba\94\93\00\d0fv6"
+    type O4 = { a: Int; foo : Int; field : Int; other : Int }; // "a\00\00\00\06\c7M\00\ba\94\93\00\d0fv6"
+
+    func go1(o : O1) : async () = async inner o;
+    func go2(o : O2) : async () = async inner o;
+    func go3(o : O3) : async () = async inner o;
```

Comment:
Why are these async? That seems irrelevant to this test, I think. Make them sync and then you can also remove the awaits below